lib/stack_master/parameter_resolvers/ami_finder.rb

```diff
@@ -0,0 +1,36 @@
+module StackMaster
+  module ParameterResolvers
+    class AmiFinder
+      def initialize(region)
+        @region = region
+        @owner = ['self']
+      end
+
+      def build_filters(value, prefix = nil)
+        owner_id = []
+        filters = value.split(',').map do |name_with_value|
+          name, value = name_with_value.strip.split('=')
+          owner_id.push value if name == 'owner_id'
+          name = prefix ? "#{prefix}:#{name}" : name
+          { name: name, values: [value] }
+        end
+        @owner = owner_id unless owner_id.empty?
+        filters
+      end
```

Comment:
Related to the earlier discussion about owner... I think the `owners` option (not the filter) sets the scope of what the filters can ... filter. Because we're setting a default of `['self']` I can see why we need to broaden the scope of owner to make the filter work.

How would you select an image owned by amazon or in the aws marketplace with this resolver? I can't see how it would be possible... Is there a way to lift the `self` restriction?